public/app/plugins/datasource/prometheus/components/PromQueryEditor.test.tsx

```diff
@@ -36,18 +50,24 @@ const setup = (propOverrides?: object) => {
 
   Object.assign(props, propOverrides);
 
-  const wrapper = shallow(<PromQueryEditor {...props} />);
-  const instance = wrapper.instance() as PromQueryEditor;
-
-  return {
-    instance,
-    wrapper,
-  };
+  return render(<PromQueryEditor {...props} />);
 };
 
 describe('Render PromQueryEditor with basic options', () => {
-  it('should render', () => {
-    const { wrapper } = setup();
-    expect(wrapper).toMatchSnapshot();
+  it('should render editor', () => {
+    setup();
+    expect(screen.getByTestId(testIds.editor)).toBeInTheDocument();
+  });
+
+  it('should render exemplar editor for dashboard', () => {
+    setup({ app: CoreApp.Dashboard });
+    expect(screen.getByTestId(testIds.editor)).toBeInTheDocument();
+    expect(screen.getByTestId(testIds.exemplar)).toBeInTheDocument();
+  });
+
+  it('should not render exemplar editor for unified alerting', () => {
+    setup({ app: CoreApp.UnifiedAlerting });
+    expect(screen.getByTestId(testIds.editor)).toBeInTheDocument();
+    expect(screen.queryByTestId(testIds.exemplar)).not.toBeInTheDocument();
```

Comment:
For user-centric react testing library tests, ideally this would read something like `expect(screen.queryByLabelText("Exemplars")).not.toBeInTheDocument()` rather than using the [test-id escape hatch](https://testing-library.com/docs/queries/bytestid).

But that might require changes larger than this (like using a Toggle for the button instead, because that's what it is 😅 ) which is fair to be out of scope. ✌🏻

Reply:
Yeah, I think that toggle would be better, and I vaguely remember that it was toggle at the beginning, but then it was change to icon button for some reasons (maybe tooltips?). But good points!